scripts/add-validation-to-crds/parse-crds.go

```diff
@@ -289,7 +289,15 @@ oneOf:
 			fields.Insert(k)
 		}
 		signature := strings.Join(sets.List(fields), ",")
-		if signature == "apiVersion,external,kind,name,namespace" {
+		if strings.HasPrefix(signature, "condition,member,memberFrom") && fieldPath == ".spec" { // IAMPolicyMember
```

Comment:
Can we make this if condition based on kind and fieldPath? `strings.HasPrefix(signature, "condition,member,memberFrom")` seem to be hacky and not super readable.

Not a blocker.

Reply:
let me take a pass on refactoring this file outside of this PR, plumbing in the kind would be helpful for all IAM CRDs' special CRD rules